src/internal/Transition.js

```diff
@@ -312,17 +327,31 @@ class Transition extends Component {
   }
 
   render() {
-    const status = this.state.status;
+    const { status } = this.state;
     if (status === UNMOUNTED) {
       return null;
     }
 
     const {
       children,
       className,
+      in: inProp,         // eslint-disable-line no-unused-vars
+      enteredClassName,   // eslint-disable-line no-unused-vars
+      enteringClassName,  // eslint-disable-line no-unused-vars
+      exitedClassName,    // eslint-disable-line no-unused-vars
+      exitingClassName,   // eslint-disable-line no-unused-vars
+      onEnter,            // eslint-disable-line no-unused-vars
+      onEntering,         // eslint-disable-line no-unused-vars
+      onEntered,          // eslint-disable-line no-unused-vars
+      onExit,             // eslint-disable-line no-unused-vars
+      onExiting,          // eslint-disable-line no-unused-vars
+      onExited,           // eslint-disable-line no-unused-vars
+      onRequestTimeout,   // eslint-disable-line no-unused-vars
+      timeout,            // eslint-disable-line no-unused-vars
+      transitionAppear,   // eslint-disable-line no-unused-vars
+      unmountOnExit,      // eslint-disable-line no-unused-vars
       ...other
     } = this.props;
-    Object.keys(propTypes).forEach((key) => delete other[key]);
```

Comment:
avoided the `delete` operator in favor of explicit removal as in other code (and for performance)